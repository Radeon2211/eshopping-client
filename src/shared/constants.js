import * as Yup from 'yup';
import { getCountries } from 'country-fns';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productConditions = {
  NEW: 'new',
  USED: 'used',
  NOT_APPLICABLE: 'not_applicable',
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([productConditions.NEW, productConditions.USED, productConditions.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
};

export const listOfAreaCodes = getCountries().map(({ name, dial }) => {
  const finalValue = `+${dial} ${name.split('(')[0].trim()}`;
  return {
    value: dial,
    label: finalValue,
  };
});

export const listOfCountries = getCountries().map(({ name }) => {
  const finalValue = name.split('(')[0].trim();
  return {
    value: finalValue,
    label: finalValue,
  };
});

export const productPages = {
  ALL_PRODUCTS: 'ALL_PRODUCTS',
  MY_PRODUCTS: 'MY_PRODUCTS',
  USER_PRODUCTS: 'USER_PRODUCTS',
};

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const orderTypes = {
  PLACED_ORDERS: 'PLACED_ORDERS',
  SELL_HISTORY: 'SELL_HISTORY',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const modalTypes = {
  ABOUT_WEBSITE: 'ABOUT_WEBSITE',
  ADD_ADMIN: 'ADD_ADMIN',
  ADD_PRODUCT: 'ADD_PRODUCT',
  BUY_PRODUCTS: 'BUY_PRODUCTS',
  CART_ITEM_ADDED: 'CART_ITEM_ADDED',
  CHANGE_NAME: 'CHANGE_NAME',
  CHANGE_EMAIL: 'CHANGE_EMAIL',
  CHANGE_USERNAME: 'CHANGE_USERNAME',
  CHANGE_PHONE_NUMBER: 'CHANGE_PHONE_NUMBER',
  CHANGE_ADDRESS: 'CHANGE_ADDRESS',
  CHANGE_PASSWORD: 'CHANGE_PASSWORD',
  CHANGE_CONTACTS: 'CHANGE_CONTACTS',
  CHANGE_DELIVERY_ADDRESS: 'CHANGE_DELIVERY_ADDRESS',
  CLEAR_CART: 'CLEAR_CART',
  DELETE_ACCOUNT: 'DELETE_ACCOUNT',
  DELETE_PRODUCT: 'DELETE_PRODUCT',
  EDIT_PRODUCT: 'EDIT_PRODUCT',
  LOGIN: 'LOGIN',
  REMOVE_ADMIN: 'REMOVE_ADMIN',
  RESET_PASSWORD: 'RESET_PASSWORD',
  PENDING_USER_INFO: 'PENDING_USER_INFO',
  SEND_VERIFICATION_LINK: 'SEND_VERIFICATION_LINK',
  SIGNUP: 'SIGNUP',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: `Price - ascending` },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: `Price - ascending` },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const updateCartActions = {
  INCREMENT: 'INCREMENT',
  DECREMENT: 'DECREMENT',
  NUMBER: 'NUMBER',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};

export const userStatuses = {
  ACTIVE: 'active',
  PENDING: 'pending',
};
