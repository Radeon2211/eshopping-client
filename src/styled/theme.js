const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#666',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#00A790',
    greenDark: '#009680',
    red: '#F22613',
  },
  shadows: {
    level1: '0 0 2px rgba(0, 0, 0, .1)',
    level2: '0 0 2px rgba(0, 0, 0, .3)',
    level3: '0 0 3px rgba(0, 0, 0, .3)',
    level4: '0 0 5px rgba(0, 0, 0, .3)',
    level5: '0 0 8px rgba(0, 0, 0, .3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Ubuntu, sans-serif',
    text: 'Lato, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
  },
};

export default theme;
