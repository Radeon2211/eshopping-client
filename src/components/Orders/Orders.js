import React, { useCallback, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useSelector, useDispatch } from 'react-redux';
import { useLocation } from 'react-router-dom';
import * as actions from '../../store/actions/indexActions';
import PlainPanel from '../UI/Panels/PlainPanel';
import InputPagination from '../Pagination/InputPagination/InputPagination';
import BottomPagination from '../Pagination/BottomPagination/BottomPagination';
import OrderList from './OrderList/OrderList';
import SortOrders from './SortOrders/SortOrders';
import Heading from '../UI/Heading/Heading';
import Loader from '../UI/Loader/Loader';
import { listItemTypes, orderTypes, defaultOrdersPerPage } from '../../shared/constants';
import * as propTypes from '../../shared/propTypes';
import { TopPagination } from '../../styled/components';

const Orders = (props) => {
  const { orders, type } = props;

  const { search } = useLocation();

  const orderCount = useSelector((state) => state.auth.orderCount);

  const dispatch = useDispatch();
  const onFetchOrders = useCallback(
    (queryParams, orderType) => dispatch(actions.fetchOrders(queryParams, orderType)),
    [dispatch],
  );

  useEffect(() => {
    onFetchOrders(search, type);
  }, [onFetchOrders, type, search]);

  let content = <Loader align="center" />;
  if (orders === null) {
    content = (
      <Heading variant="h4" align="center" data-testid="Orders-error">
        {type === orderTypes.PLACED_ORDERS
          ? 'There is a problem to fetch your placed orders'
          : 'There is a problem to fetch your sell history'}
      </Heading>
    );
  } else if (orders) {
    if (orders.length <= 0) {
      content = (
        <Heading variant="h4" align="center" data-testid="Orders-no-orders-info">
          {type === orderTypes.PLACED_ORDERS
            ? `You don't have any placed orders yet`
            : 'Your sell history is empty'}
        </Heading>
      );
    } else if (orderCount > 0) {
      content = (
        <PlainPanel>
          <TopPagination>
            <SortOrders />
            <InputPagination itemQuantity={orderCount} quantityPerPage={defaultOrdersPerPage} />
          </TopPagination>
          <OrderList orders={orders} orderType={type} />
          <BottomPagination
            itemQuantity={orderCount}
            itemsType={listItemTypes.ORDER}
            quantityPerPage={defaultOrdersPerPage}
          />
        </PlainPanel>
      );
    }
  }

  return content;
};

Orders.propTypes = {
  orders: PropTypes.arrayOf(PropTypes.shape(propTypes.orderItem)),
  type: PropTypes.string.isRequired,
};

export default Orders;
