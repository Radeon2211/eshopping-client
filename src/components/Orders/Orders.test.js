import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import theme from '../../styled/theme';
import Orders from './Orders';
import {
  checkProps,
  createOrder,
  createTransactionAndOrderProdItem,
} from '../../shared/testUtility/testUtility';
import { orderTypes } from '../../shared/constants';

const mockStore = configureMockStore([thunk]);

const defaultHistory = {
  listen: jest.fn(),
  createHref: jest.fn(),
  location: { pathname: '/my-account/placed-orders', search: '?p=1' },
};

const setUp = (orders, orderCount, type = orderTypes.PLACED_ORDERS) => {
  const store = mockStore({
    auth: { orderCount },
    ui: { isDataLoading: false },
  });

  return render(
    <Router history={defaultHistory}>
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Orders orders={orders} type={type} />
        </ThemeProvider>
      </Provider>
    </Router>,
  );
};

afterEach(cleanup);

describe('<Orders />', () => {
  describe('Check prop types', () => {
    it('should NOT throw a warning', () => {
      const expectedProps = {
        orders: [createOrder()],
        type: orderTypes.PLACED_ORDERS,
      };
      expect(checkProps(Orders, expectedProps)).toBeUndefined();
    });

    it('should throw a warning', () => {
      expect(checkProps(Orders, {})).not.toBeNull();
    });
  });

  describe('Check how renders', () => {
    it('should render only <Loader />', () => {
      const { asFragment } = setUp(undefined, undefined);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should everything correctly with one order and type PLACED_ORDERS', () => {
      const products = [
        createTransactionAndOrderProdItem({
          productId: 'p1',
          sellerUsername: 'sellerUser',
          price: 4,
          quantity: 4,
          name: 'product1',
          photo: true,
        }),
      ];
      const orders = [
        createOrder({
          products,
          id: 'o1',
          sellerUsername: 'sellerUser',
          buyerUsername: 'buyerUser',
          overallPrice: 16,
          createdAt: '2021-01-08T11:08:51.008Z',
        }),
      ];
      const { asFragment } = setUp(orders, 1, orderTypes.PLACED_ORDERS);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should everything correctly with two orders and type SELL_HISTORY', () => {
      const products1 = [
        createTransactionAndOrderProdItem({
          productId: 'p1',
          sellerUsername: 'sellerUser',
          price: 4,
          quantity: 4,
          name: 'product1',
          photo: true,
        }),
      ];
      const products2 = [
        createTransactionAndOrderProdItem({
          productId: 'p2',
          sellerUsername: 'sellerUser',
          price: 6.4,
          quantity: 2,
          name: 'product2',
          photo: true,
        }),
      ];
      const orders = [
        createOrder({
          products: products1,
          id: 'o1',
          sellerUsername: 'sellerUser',
          buyerUsername: 'buyerUser',
          overallPrice: 16,
          createdAt: '2021-01-08T11:08:51.008Z',
        }),
        createOrder({
          products: products2,
          id: 'o2',
          sellerUsername: 'sellerUser',
          buyerUsername: 'buyerUser',
          overallPrice: 12.8,
          createdAt: '2021-02-18T21:12:35.008Z',
        }),
      ];
      const { asFragment } = setUp(orders, 2, orderTypes.SELL_HISTORY);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render correct info if orders are null and type PLACED_ORDERS', () => {
      const { asFragment } = setUp(null, undefined, orderTypes.PLACED_ORDERS);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render correct info if orders are empty array and type PLACED_ORDERS', () => {
      const { asFragment } = setUp([], undefined, orderTypes.PLACED_ORDERS);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render correct info if orders are null and type SELL_HISTORY', () => {
      const { asFragment } = setUp(null, undefined, orderTypes.SELL_HISTORY);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render correct info if orders are empty array and type SELL_HISTORY', () => {
      const { asFragment } = setUp([], undefined, orderTypes.SELL_HISTORY);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
