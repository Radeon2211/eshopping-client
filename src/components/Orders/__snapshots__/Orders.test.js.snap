// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Orders /> Check how renders should everything correctly with one order and type PLACED_ORDERS 1`] = `
<DocumentFragment>
  <section
    class="sc-bdfBwQ dIpFeA"
  >
    <div
      class="sc-kEjbxe dSjtmZ"
    >
      <div
        class="sc-idOhPF ckvBxo"
      >
        <label
          class="label"
          for="sortOption"
        >
          Sort
        </label>
        <div
          class="select css-2b097c-container"
          id="sortOption"
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-l640om-control"
          >
            <div
              class=" css-g1d714-ValueContainer"
            >
              <div
                class=" css-1uccc91-singleValue"
              >
                Date - from newest
              </div>
              <input
                aria-autocomplete="list"
                class="css-wmatm6-dummyInput-DummyInput"
                id="react-select-2-input"
                readonly=""
                tabindex="0"
                value=""
              />
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-ml9fjp-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-gsTCUz hXFboD"
        data-testid="InputPagination"
      >
        <a
          class="arrow hide-arrow"
          data-testid="InputPagination-left-arrow"
        >
          <div
            class="sc-dlfnbm jOFrrM"
            color=""
          >
            <svg>
              arrow.svg
            </svg>
          </div>
        </a>
        <form
          class="form-number"
        >
          <input
            class="sc-eCssSg gPPIII"
            data-testid="NumberInput-page"
            name="page"
            type="number"
            value="1"
          />
        </form>
        <span
          class="sc-hKgILt gvqqww"
          color=""
          display="inline-block"
          overflow="visible"
          size="3"
          spacing="0px"
          transform=""
        >
          of
        </span>
        <span
          class="sc-hKgILt gvqqww"
          color=""
          display="inline-block"
          overflow="visible"
          size="3"
          spacing="0px"
          transform=""
        >
          1
        </span>
        <a
          class="arrow hide-arrow"
          data-testid="InputPagination-right-arrow"
        >
          <div
            class="sc-dlfnbm lebiLC"
            color=""
          >
            <svg>
              arrow.svg
            </svg>
          </div>
        </a>
      </div>
    </div>
    <div
      class="sc-iBPRYJ iceOJm"
    >
      <div
        class="sc-fubCfw cijXbO"
      >
        <div
          class="sc-iqHYGH fAChvl"
          direction="row"
          spacing="3"
          width="100%"
          wrap="nowrap"
        >
          <span
            class="sc-hKgILt fttxOE"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            <span>
              seller 
            </span>
            <a
              data-testid="OrderList-user-link"
            >
              <span
                class="sc-pFZIQ kpCSTb"
              >
                sellerUser
              </span>
            </a>
          </span>
          <span
            class="sc-hKgILt cTDtWF"
            color=""
            display="inline-block"
            overflow="visible"
            size="2"
            spacing="0px"
            transform=""
          >
            8 Jan 2021, 12:08
          </span>
        </div>
        <div
          class="sc-iqHYGH ZKsJN sc-crrsfI jEsAbF"
          data-testid="TransactionAndOrderProdItem"
          direction="row"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <a
            data-testid="TransactionAndOrderProdItem-product-link-photo"
          >
            <div
              class="sc-iqHYGH fVPTNg sc-kstrdz hcpuVO"
              data-testid="ProductThumbnail"
              direction="row"
              height="5"
              spacing=""
              width="5"
              wrap="nowrap"
            >
              <img
                alt="product1"
                data-testid="ProductThumbnail-img"
                src="http://192.168.1.109:4000/orders/o1/p1/photo"
              />
            </div>
          </a>
          <div
            class="sc-dQppl cZhraS"
          >
            <span
              class="name"
            >
              <a
                data-testid="TransactionAndOrderProdItem-product-link-name"
              >
                product1
              </a>
            </span>
            <div
              class="sc-iqHYGH bALALa sc-bqyKva kIpkqx"
              direction="row"
              spacing=""
              width="100%"
              wrap="nowrap"
            >
              <span
                class="sc-hKgILt jQArHJ"
                color="#777"
                display="inline-block"
                overflow="visible"
                size="2"
                spacing="0px"
                transform=""
              >
                4 x $4
              </span>
              <span
                class="sc-hKgILt dKfWou"
                color=""
                display="inline-block"
                overflow="visible"
                size="4"
                spacing="0px"
                transform=""
              >
                $16
              </span>
            </div>
          </div>
        </div>
        <div
          class="sc-iqHYGH kRDxPF"
          direction="row"
          spacing="3"
          width="100%"
          wrap="nowrap"
        >
          <a
            data-testid="OrderList-order-details-link"
          >
            <button
              class="sc-hBEYos cmoUln"
              color="blue"
              type="button"
            >
              details
            </button>
          </a>
          <div>
            <span
              class="sc-hKgILt cmAqUt"
              color="#777"
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="1px"
              transform=""
            >
              TOTAL
            </span>
            <span
              class="sc-hKgILt kRfSFE"
              color=""
              display="inline-block"
              overflow="visible"
              size="5"
              spacing="0px"
              transform=""
            >
              $16
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-gKsewC fGOzvo"
    >
      <span
        class="sc-hKgILt kshmRe"
        color="#777"
        display="inline-block"
        overflow="visible"
        size="1"
        spacing="0px"
        transform=""
      >
        1 - 1 of 1 order
      </span>
      <div
        class="sc-jSgupP gFFAhR"
        data-testid="NumberPagination"
      >
        <a
          class="arrow hide-arrow"
          data-testid="NumberPagination-left-arrow"
        >
          <div
            class="sc-dlfnbm jOFrrM"
            color=""
          >
            <svg>
              arrow.svg
            </svg>
          </div>
        </a>
        <a
          class="number-link active"
          data-testid="NumberPagination-page1"
        >
          1
        </a>
        <span
          class="of"
        >
          of
        </span>
        <span
          class="of"
        >
          1
        </span>
        <a
          class="arrow hide-arrow"
          data-testid="NumberPagination-right-arrow"
        >
          <div
            class="sc-dlfnbm lebiLC"
            color=""
          >
            <svg>
              arrow.svg
            </svg>
          </div>
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`<Orders /> Check how renders should everything correctly with two orders and type SELL_HISTORY 1`] = `
<DocumentFragment>
  <section
    class="sc-bdfBwQ dIpFeA"
  >
    <div
      class="sc-kEjbxe dSjtmZ"
    >
      <div
        class="sc-idOhPF ckvBxo"
      >
        <label
          class="label"
          for="sortOption"
        >
          Sort
        </label>
        <div
          class="select css-2b097c-container"
          id="sortOption"
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-l640om-control"
          >
            <div
              class=" css-g1d714-ValueContainer"
            >
              <div
                class=" css-1uccc91-singleValue"
              >
                Date - from newest
              </div>
              <input
                aria-autocomplete="list"
                class="css-wmatm6-dummyInput-DummyInput"
                id="react-select-3-input"
                readonly=""
                tabindex="0"
                value=""
              />
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-ml9fjp-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-gsTCUz hXFboD"
        data-testid="InputPagination"
      >
        <a
          class="arrow hide-arrow"
          data-testid="InputPagination-left-arrow"
        >
          <div
            class="sc-dlfnbm jOFrrM"
            color=""
          >
            <svg>
              arrow.svg
            </svg>
          </div>
        </a>
        <form
          class="form-number"
        >
          <input
            class="sc-eCssSg gPPIII"
            data-testid="NumberInput-page"
            name="page"
            type="number"
            value="1"
          />
        </form>
        <span
          class="sc-hKgILt gvqqww"
          color=""
          display="inline-block"
          overflow="visible"
          size="3"
          spacing="0px"
          transform=""
        >
          of
        </span>
        <span
          class="sc-hKgILt gvqqww"
          color=""
          display="inline-block"
          overflow="visible"
          size="3"
          spacing="0px"
          transform=""
        >
          1
        </span>
        <a
          class="arrow hide-arrow"
          data-testid="InputPagination-right-arrow"
        >
          <div
            class="sc-dlfnbm lebiLC"
            color=""
          >
            <svg>
              arrow.svg
            </svg>
          </div>
        </a>
      </div>
    </div>
    <div
      class="sc-iBPRYJ iceOJm"
    >
      <div
        class="sc-fubCfw cijXbO"
      >
        <div
          class="sc-iqHYGH fAChvl"
          direction="row"
          spacing="3"
          width="100%"
          wrap="nowrap"
        >
          <span
            class="sc-hKgILt fttxOE"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            <span>
              buyer 
            </span>
            <a
              data-testid="OrderList-user-link"
            >
              <span
                class="sc-pFZIQ kpCSTb"
              >
                buyerUser
              </span>
            </a>
          </span>
          <span
            class="sc-hKgILt cTDtWF"
            color=""
            display="inline-block"
            overflow="visible"
            size="2"
            spacing="0px"
            transform=""
          >
            8 Jan 2021, 12:08
          </span>
        </div>
        <div
          class="sc-iqHYGH ZKsJN sc-crrsfI jEsAbF"
          data-testid="TransactionAndOrderProdItem"
          direction="row"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <a
            data-testid="TransactionAndOrderProdItem-product-link-photo"
          >
            <div
              class="sc-iqHYGH fVPTNg sc-kstrdz hcpuVO"
              data-testid="ProductThumbnail"
              direction="row"
              height="5"
              spacing=""
              width="5"
              wrap="nowrap"
            >
              <img
                alt="product1"
                data-testid="ProductThumbnail-img"
                src="http://192.168.1.109:4000/orders/o1/p1/photo"
              />
            </div>
          </a>
          <div
            class="sc-dQppl cZhraS"
          >
            <span
              class="name"
            >
              <a
                data-testid="TransactionAndOrderProdItem-product-link-name"
              >
                product1
              </a>
            </span>
            <div
              class="sc-iqHYGH bALALa sc-bqyKva kIpkqx"
              direction="row"
              spacing=""
              width="100%"
              wrap="nowrap"
            >
              <span
                class="sc-hKgILt jQArHJ"
                color="#777"
                display="inline-block"
                overflow="visible"
                size="2"
                spacing="0px"
                transform=""
              >
                4 x $4
              </span>
              <span
                class="sc-hKgILt dKfWou"
                color=""
                display="inline-block"
                overflow="visible"
                size="4"
                spacing="0px"
                transform=""
              >
                $16
              </span>
            </div>
          </div>
        </div>
        <div
          class="sc-iqHYGH kRDxPF"
          direction="row"
          spacing="3"
          width="100%"
          wrap="nowrap"
        >
          <a
            data-testid="OrderList-order-details-link"
          >
            <button
              class="sc-hBEYos cmoUln"
              color="blue"
              type="button"
            >
              details
            </button>
          </a>
          <div>
            <span
              class="sc-hKgILt cmAqUt"
              color="#777"
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="1px"
              transform=""
            >
              TOTAL
            </span>
            <span
              class="sc-hKgILt kRfSFE"
              color=""
              display="inline-block"
              overflow="visible"
              size="5"
              spacing="0px"
              transform=""
            >
              $16
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-fubCfw cijXbO"
      >
        <div
          class="sc-iqHYGH fAChvl"
          direction="row"
          spacing="3"
          width="100%"
          wrap="nowrap"
        >
          <span
            class="sc-hKgILt fttxOE"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            <span>
              buyer 
            </span>
            <a
              data-testid="OrderList-user-link"
            >
              <span
                class="sc-pFZIQ kpCSTb"
              >
                buyerUser
              </span>
            </a>
          </span>
          <span
            class="sc-hKgILt cTDtWF"
            color=""
            display="inline-block"
            overflow="visible"
            size="2"
            spacing="0px"
            transform=""
          >
            18 Feb 2021, 22:12
          </span>
        </div>
        <div
          class="sc-iqHYGH ZKsJN sc-crrsfI jEsAbF"
          data-testid="TransactionAndOrderProdItem"
          direction="row"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <a
            data-testid="TransactionAndOrderProdItem-product-link-photo"
          >
            <div
              class="sc-iqHYGH fVPTNg sc-kstrdz hcpuVO"
              data-testid="ProductThumbnail"
              direction="row"
              height="5"
              spacing=""
              width="5"
              wrap="nowrap"
            >
              <img
                alt="product2"
                data-testid="ProductThumbnail-img"
                src="http://192.168.1.109:4000/orders/o2/p2/photo"
              />
            </div>
          </a>
          <div
            class="sc-dQppl cZhraS"
          >
            <span
              class="name"
            >
              <a
                data-testid="TransactionAndOrderProdItem-product-link-name"
              >
                product2
              </a>
            </span>
            <div
              class="sc-iqHYGH bALALa sc-bqyKva kIpkqx"
              direction="row"
              spacing=""
              width="100%"
              wrap="nowrap"
            >
              <span
                class="sc-hKgILt jQArHJ"
                color="#777"
                display="inline-block"
                overflow="visible"
                size="2"
                spacing="0px"
                transform=""
              >
                2 x $6.40
              </span>
              <span
                class="sc-hKgILt dKfWou"
                color=""
                display="inline-block"
                overflow="visible"
                size="4"
                spacing="0px"
                transform=""
              >
                $12.80
              </span>
            </div>
          </div>
        </div>
        <div
          class="sc-iqHYGH kRDxPF"
          direction="row"
          spacing="3"
          width="100%"
          wrap="nowrap"
        >
          <a
            data-testid="OrderList-order-details-link"
          >
            <button
              class="sc-hBEYos cmoUln"
              color="blue"
              type="button"
            >
              details
            </button>
          </a>
          <div>
            <span
              class="sc-hKgILt cmAqUt"
              color="#777"
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="1px"
              transform=""
            >
              TOTAL
            </span>
            <span
              class="sc-hKgILt kRfSFE"
              color=""
              display="inline-block"
              overflow="visible"
              size="5"
              spacing="0px"
              transform=""
            >
              $12.80
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-gKsewC fGOzvo"
    >
      <span
        class="sc-hKgILt kshmRe"
        color="#777"
        display="inline-block"
        overflow="visible"
        size="1"
        spacing="0px"
        transform=""
      >
        1 - 2 of 2 orders
      </span>
      <div
        class="sc-jSgupP gFFAhR"
        data-testid="NumberPagination"
      >
        <a
          class="arrow hide-arrow"
          data-testid="NumberPagination-left-arrow"
        >
          <div
            class="sc-dlfnbm jOFrrM"
            color=""
          >
            <svg>
              arrow.svg
            </svg>
          </div>
        </a>
        <a
          class="number-link active"
          data-testid="NumberPagination-page1"
        >
          1
        </a>
        <span
          class="of"
        >
          of
        </span>
        <span
          class="of"
        >
          1
        </span>
        <a
          class="arrow hide-arrow"
          data-testid="NumberPagination-right-arrow"
        >
          <div
            class="sc-dlfnbm lebiLC"
            color=""
          >
            <svg>
              arrow.svg
            </svg>
          </div>
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`<Orders /> Check how renders should render correct info if orders are empty array and type PLACED_ORDERS 1`] = `
<DocumentFragment>
  <h4
    class="sc-dIUggk qbmUV"
  >
    You don't have any placed orders yet
  </h4>
</DocumentFragment>
`;

exports[`<Orders /> Check how renders should render correct info if orders are empty array and type SELL_HISTORY 1`] = `
<DocumentFragment>
  <h4
    class="sc-dIUggk qbmUV"
  >
    Your sell history is empty
  </h4>
</DocumentFragment>
`;

exports[`<Orders /> Check how renders should render correct info if orders are null and type PLACED_ORDERS 1`] = `
<DocumentFragment>
  <h4
    class="sc-dIUggk qbmUV"
  >
    There is a problem to fetch your placed orders
  </h4>
</DocumentFragment>
`;

exports[`<Orders /> Check how renders should render correct info if orders are null and type SELL_HISTORY 1`] = `
<DocumentFragment>
  <h4
    class="sc-dIUggk qbmUV"
  >
    There is a problem to fetch your sell history
  </h4>
</DocumentFragment>
`;

exports[`<Orders /> Check how renders should render only <Loader /> 1`] = `
<DocumentFragment>
  <div
    class="sc-fodVxV eaPFxd"
    data-testid="LoaderWrapper"
  >
    <div
      class="sc-fFubgz cHAJHi"
      data-testid="Loader"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</DocumentFragment>
`;
